connexion/operation.py

```diff
@@ -89,10 +87,34 @@ def __init__(self, method, path, operation, app_produces, app_security,
         self.security = operation.get('security', app_security)
         self.__undecorated_function = resolver(self.operation_id)
 
-    def detect_controller(self, operation_id, router_controller):
-        if router_controller is None:
+    def detect_controller(self):
+
+        operation_id = self.operation.get('operationId')
+        x_router_controller = self.operation.get('x-swagger-router-controller')
+
+        if operation_id:
+            if x_router_controller:
+                return x_router_controller + '.' + operation_id
             return operation_id
-        return router_controller + '.' + operation_id
+
+        function = self.method.lower()
+
+        if x_router_controller:
+            mod_name = x_router_controller
+        else:
+            if not self.default_module_name:
+                raise InvalidSpecification(
+                    "Neither operationId or x-swagger-router-controller was "
+                    + "configured and no default module set for Api"
+                )
+
+            mod_name = self.default_module_name
+            match = re.search('^/(.+?)(/.?|$)', self.path)
+            if match:
+                mod_name += '.' + match.group(1)
+                function = ('search' if function == 'get' and match.group(2).strip('/') == '' else function)
```

Comment:
Well, that's what you get when you use defaults over configuration. Sure it's arguable, but nobody is stopping no one from configuring an operationId where you want something different, even if that's 'everywhere'. It just becomes *optional* exactly as the Swagger spec requires.